### PR TITLE
nmap: fixes for 32-bit and pre-C++11 platforms

### PR DESCRIPTION
**Pull request by chrstphrchvz**
Fixes: https://trac.macports.org/ticket/68543

###### Type(s)

- [x] bugfix

###### Tested on

Not tested on affected macOS versions.
32-bit fix tested on IA-32 Linux; only eliminates a compiler warning on 64-bit platforms.

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: net/nmap/Portfile
Comment (chrstphrchvz) [inline net/nmap/Portfile:7]: There is no change on 64-bit except for a compiler warning going away. I could propose incrementing the revision only on macOS versions for 32-bit processors, but I have not done so to avoid confusion.

File: net/nmap/Portfile
Comment (chrstphrchvz) [inline net/nmap/Portfile:36]: Because this patch could be accepted upstream, I have prefixed it with `0001…` so that it is applied before the `0002…` patch.

**Comment by pmetzger** [comment]
FWIW, I'm not clear on why this isn't openmaintainer.

**Comment by danielluke** [comment]
> FWIW, I'm not clear on why this isn't openmaintainer.

Because I don't prefer openmaintainer on ports that I maintain.

**Review by danielluke — CHANGES_REQUESTED** [review]

File: net/nmap/Portfile
Comment (chrstphrchvz) [reply net/nmap/Portfile]: I suggest this patch be applied unconditionally, since the patch has no undesirable effect on non-GCC 4.2 platforms, and so that port updaters make sure this patch still works regardless of their platform.

**Comment by pmetzger** [comment]
@danielluke May we commit?

**Comment by ghosthound** [comment]
I don't have anything 32bit to test this with either so can't do much other than look at the changes and note that the automated tests pass.  I'm good with the changes, would like to have @danielluke approve as well since he's been involved more than I have.

**Comment by barracuda156** [comment]
We also need to define `_STDC_FORMAT_MACROS`, I guess:
```
MACLookup.cc: In function ‘void mac_prefix_init()’:
MACLookup.cc:161: error: expected `)' before ‘PRIX64’
MACLookup.cc:161: warning: conversion lacks type at end of format
MACLookup.cc:161: warning: too many arguments for format
make: *** [MACLookup.o] Error 1
```

**Comment by barracuda156** [comment]
@chrstphrchvz `nbase/nbase.h` should have the following above other headers there:
```
#ifndef __STDC_FORMAT_MACROS
#define __STDC_FORMAT_MACROS
#endif
```

**Comment by chrstphrchvz** [comment]
> @chrstphrchvz `nbase/nbase.h` should have the following above other headers there:
> 
> ```
> #ifndef __STDC_FORMAT_MACROS
> #define __STDC_FORMAT_MACROS
> #endif
> ```

I have added this to just before `#include <inttypes.h>` since that should be the only header which would recognize this macro.

**Comment by barracuda156** [comment]
> > @chrstphrchvz `nbase/nbase.h` should have the following above other headers there:
> > ```
> > #ifndef __STDC_FORMAT_MACROS
> > #define __STDC_FORMAT_MACROS
> > #endif
> > ```
> 
> I have added this to just before `#include <inttypes.h>` since that should be the only header which would recognize this macro.

@chrstphrchvz I did exactly that at first, and it failed. I think something else which comes above pulls in the same header.

I can rerun the build that way again just to make sure it was not something random.

File: net/nmap/Portfile
Comment (danielluke) [inline net/nmap/Portfile:37]: I'd still prefer the gcc-4.2 patch to be only applied in platform blocks that restrict it to 10.5 and earlier. Hopefully upstream will include the 32bit patch (or equivalent) and we can just keep this patching only on older hosts that actually need it.

**Review by danielluke — CHANGES_REQUESTED** [review]

File: net/nmap/Portfile
Comment (barracuda156) [reply net/nmap/Portfile:37]: gcc-4.2 is a part of Xcode on 10.6 too, so any patches targeting it should extend to 10.6.

**Comment by barracuda156** [comment]
Do we have provisionally final thing? I can test tomorrow afternoon.

**Comment by barracuda156** [comment]
@pmetzger Let me test once again just to be sure. Will update in half an hour.

**Comment by danielluke** [comment]
@pmetzger please follow the normal process for PR merging in the future (ie, non-openmaintainer ports wait for maintainer approval or require the maintainer timeout process).

**Pull request merged**
